tests/test_structure_manager_centers.cc

```diff
@@ -40,23 +40,31 @@ namespace rascal {
   // checking iteration
   BOOST_FIXTURE_TEST_CASE(iterator_test,
                           ManagerFixture<StructureManagerCenters>) {
-    BOOST_CHECK_EQUAL(manager->get_size(), atom_types.size());
-    BOOST_CHECK_EQUAL(manager->get_nb_clusters(1), atom_types.size());
-
-    int atom_counter{};
     constexpr bool verbose{false};
-
-    for (auto atom_cluster : manager) {
-      BOOST_CHECK_EQUAL(atom_counter, atom_cluster.get_index());
-      ++atom_counter;
-      for (int ii{3}; ii < 3; ++ii) {
-        BOOST_CHECK_EQUAL(positions(ii, atom_cluster.get_index()),
-                          atom_cluster.get_position()[ii]);
-      }
-      if (verbose) {
-        std::cout << "atom (" << atom_cluster.back()
-                  << "), atom counter = " << atom_counter << std::endl;
+    int i_manager{0};
+    for (auto & manager : this->managers) {
+      auto & structure = this->structures[i_manager];
+      auto & is_a_center_atom = structure.is_a_center_atom;
+      BOOST_CHECK_EQUAL(manager->get_size(), is_a_center_atom.count());
```

Comment:
I need to think about this a little bit more, `get_size()` is needed for iteration, and you only want to iterate over the _marked_ centers. So it makes total sense here. But on the other hand I still find it unusual. Maybe it is the naming (mask instead of nocenter) that would help me wrap my head around it.